src/main/java/com/redhat/devtools/lsp4ij/usages/LSPExternalReferencesFinder.java

```diff
@@ -0,0 +1,172 @@
+                            PsiReferenceUtil.unwrapMultiReference(originalReference) :
+                            Collections.emptyList();
+                    for (PsiReference reference : references) {
+                        // Deduplicate using a unique key with reference type, file, and text range
+                        String referenceKey = getReferenceKey(reference);
+                        if (referenceKey != null) {
+                            // Only add references we haven't added previously
+                            if (!externalReferenceKeys.contains(referenceKey)) {
+                                PsiElement targetElement = reference.resolve();
+                                PsiFile targetFile = targetElement != null ? targetElement.getContainingFile() : null;
+                                // Files match
+                                if ((targetFile != null) && Objects.equals(file, targetFile)) {
+                                    // Text ranges match
+                                    TextRange targetTextRange = targetElement.getTextRange();
+                                    if ((targetTextRange != null) && Objects.equals(wordTextRange, targetTextRange)) {
+                                        // Text matches according to case-sensitivity
+                                        String targetText = reference.getCanonicalText();
+                                        if (caseSensitive ? wordText.equals(targetText) : wordText.equalsIgnoreCase(targetText)) {
+                                            if (!processor.process(reference)) {
+                                                return false;
+                                            }
+                                            externalReferenceKeys.add(referenceKey);
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                    if (progressIndicator != null) {
+                        progressIndicator.checkCanceled();
+                    }
+                    return true;
+                },
+                ReadAction.compute(file::getUseScope),
+                wordText,
+                UsageSearchContext.ANY,
+                caseSensitive
+        );
+    }
+
+    private static boolean isCaseSensitive(@NotNull Set<LanguageServerWrapper> languageServerWrappers, @NotNull PsiFile psiFile) {
+        // If any supporting language server is case-sensitive, the search must be case-sensitive; it's better to
+        // miss changing things that should have been changed than to change things that should not
+        for (LanguageServerWrapper languageServerWrapper : languageServerWrappers) {
+            LSPClientFeatures clientFeatures = languageServerWrapper.getClientFeatures();
+            if ((clientFeatures != null) && clientFeatures.isCaseSensitive(psiFile)) {
```

Comment:
clientFeatures != null should never happen

Reply:
I'll annotate that method as `@NotNull` in the commit that removes that check so that there's no confusion going forward.